airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -178,19 +170,11 @@ def parse_date(self, date: str) -> datetime.datetime:
         return self._parser.parse(date, self.datetime_format, self._timezone)
 
     @classmethod
-    def _parse_timedelta(cls, time_str):
+    def _parse_timedelta(cls, time_str) -> Union[datetime.timedelta, Duration]:
         """
-        Parse a time string e.g. (2h13m) into a timedelta object.
-        Modified from virhilo's answer at https://stackoverflow.com/a/4628148/851699
-        :param time_str: A string identifying a duration. (eg. 2h13m)
-        :return relativedelta: A relativedelta object
+        :return Parses an ISO 8601 durations into datetime.timedelta or Duration objects.
         """
-        parts = cls.timedelta_regex.match(time_str)
-
-        assert parts is not None
-
-        time_params = {name: float(param) for name, param in parts.groupdict().items() if param}
-        return relativedelta(**time_params)
+        return parse_duration(time_str)
```

Comment:
Can we find something with higher adoption that does the same sort of thing? Feels like we want to be careful introducing dependencies that aren't widely used. 

ETA: doing a quick search I didn't really see an alternative for parsing ISO duration. I wonder at what point it's better to use our own rather than introducing a dependency.

Reply:
I had the same question. I don't understand how parsing ISO8601 duration is not part of the standard Python library (my guess is that the standard library does not support timedelta for months and years but eh..)

Using `relativedelta` to support years and months, we would only have the parsing of the string to do. I don't have information to confirm it but my guts feeling was that it's less work to use this library than to write the parsing and test it. My reasoning was that there are 29,9k users for this lib which seems like a good enough community to battle test the code. If you have information that would suggest otherwise, please share them with me and I'll change that.

Reply:
Makes sense to me, and I agree with you that 29.9k users seems legit - I'd overlooked that (was looking at number of stars).